### PR TITLE
bugfix the three vertices are collinear

### PR DESCRIPTION
**Pull request by Jackie9527**
The three vertices are collinear when converting a polygon to a triangle, which is weird but indeed exists.
Currently, we just skip such ear.

### CODE REVIEW

File: code/PostProcessing/TriangulateProcess.cpp [hunk 1]
```diff
@@ -468,6 +468,20 @@ bool TriangulateProcess::TriangulateMesh( aiMesh* pMesh)
...
+                    aiVector2D left = *pnt0 - *pnt1;
+                    aiVector2D right = *pnt2 - *pnt1;
+
+                    left.Normalize();
+                    right.Normalize();
+                    auto mul = left * right;
+
+                    // if the angle is 0 or 180
+                    if (std::abs(mul - 1.f) < ai_epsilon || std::abs(mul + 1.f) < ai_epsilon) {
+                        // skip this ear
```
Comment (turol) [inline code/PostProcessing/TriangulateProcess.cpp:481]: This should log a warning so user knows the model is potentially broken.

Comment (Jackie9527) [reply code/PostProcessing/TriangulateProcess.cpp:481]: done

**Review by kimkulling — APPROVED** [review]
Looks fine.

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**